packages/core/src/fhir-to-cda/cda-templates/components/augmented-resources.ts

```diff
@@ -5,26 +5,31 @@ import {
   Observation,
   Resource,
 } from "@medplum/fhirtypes";
+import { oids } from "../constants";
 
-export interface AugmentedResource {
+export interface AugmentedResource<T extends Resource> {
   readonly typeOid: string;
   readonly sectionName: string;
-  resource: Resource;
+  resource: T;
 }
-export class AugmentedObservation implements AugmentedResource {
-  constructor(public typeOid: string, public sectionName: string, public resource: Observation) {}
+export class AugmentedObservation implements AugmentedResource<Observation> {
+  constructor(
+    public readonly typeOid: string,
```

Comment:
nit: make `typeOid` the 3rd param so the first two match the other "augmented
 classes?